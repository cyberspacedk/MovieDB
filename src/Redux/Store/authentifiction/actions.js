/* eslint-disable no-shadow */
/* eslint-disable prettier/prettier */
import axios from 'axios';

// ASYNC AWAIT SESSION ID
/*
const authentification = async (
  reqTokenPath,
  verifyPath,
  username,
  password,
  sessionPath,
) => {
  const requestToken = await axios(reqTokenPath).then(
    ({ data }) => data.request_token,
  );
  console.log(requestToken);

  const verifiedToken = await axios
    .post(verifyPath, {
      username,
      password,
      request_token: requestToken,
    })
    .then(({ data }) => data.request_token);

  const sessionId = await axios
    .post(sessionPath, { request_token: verifiedToken })
    .then(({ data }) => data.session_id);

  console.log(sessionId);
};
*/

const newRequestToken = reqTokenPath => {
  axios(reqTokenPath).then(({ data }) =>
    localStorage.setItem('REQUEST_TOKEN', JSON.stringify(data.request_token)),
  );
};

const authentification = (
  tokenPath,
  username,
  password,
  sessionPath,
) => dispatch => {
  axios
    .post(tokenPath, {
      username,
      password,
      request_token: JSON.parse(localStorage.getItem('REQUEST_TOKEN')),
    })
    .then(({ data }) => {
      axios
        .post(sessionPath, { request_token: data.request_token })
        .then(({ data }) => {
          dispatch({
            type: 'SET_SESSION_ID',
            payload: data.session_id,
          });
          dispatch({
            type: 'GET_USERLOGIN',
            payload: username,
          });

          localStorage.setItem('SESSION_ID', JSON.stringify(data.session_id));
          localStorage.setItem('USERNAME', JSON.stringify(username));
        })
        .catch(() => alert('Bad token !!!'));
    })
    .catch(err => console.log(err));
};

const userLogout = (deletePath, sessionId) => dispatch => {
  axios
    .delete(deletePath, {
      data: {
        session_id: sessionId,
      },
    })
    .then(() => {
      dispatch({
        type: 'DELETE_SESSION_ID',
      });
      localStorage.clear();
    })
    .catch(err => console.log(err));
};

export { authentification, newRequestToken, userLogout };
