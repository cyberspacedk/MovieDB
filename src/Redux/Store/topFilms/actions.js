export const fetchDataRequest = () => ({ type: 'FETCH_REQUEST' });

export const fetchDataCancel = () => ({ type: 'FETCH_CANCEL' });

export const fetchDataError = error => ({
  type: 'FETCH_ERROR',
  payload: error,
});

export const fetchDataSuccess = data => ({
  type: 'FETCH_RESPONSE',
  payload: data,
});
