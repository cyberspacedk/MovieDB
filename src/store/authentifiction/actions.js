const authRequest = ({ username, password }) => ({
  type: 'AUTH_REQUEST',
  payload: {
    username,
    password,
  },
});

const authSuccess = userData => ({ type: 'AUTH_SUCCESS', payload: userData });

const authError = () => ({ type: 'AUTH_ERROR' });

const authLogout = () => ({ type: 'AUTH_LOGOUT' });

const tryAgain = () => ({
  type: 'TRY_AGAIN',
});

export { authLogout, authRequest, authSuccess, authError, tryAgain };
