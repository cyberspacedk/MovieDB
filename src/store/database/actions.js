const writeToDatabase = ({ movies = {}, lists = {}, genres = {} }) => ({
  type: 'WRITE_TO_DATABASE',
  payload: {
    movies,
    lists,
    genres,
  },
});

export default writeToDatabase;
