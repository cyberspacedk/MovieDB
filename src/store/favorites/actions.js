const operationsFavoritesRequest = (movieId, what) => ({
  type: 'OPERATIONS_FAVORITES_REQUEST',
  payload: movieId,
  whatToDo: what,
});

const getFavoritesError = () => ({
  type: 'GET_FAVORITES_ERROR',
});

const getFavoritesRequest = (page = 1) => ({
  type: 'GET_FAVORITES_REQUEST',
  payload: page,
});

const getFavoritesResponse = fav => ({
  type: 'GET_FAVORITES_RESPONSE',
  payload: fav,
});

export {
  operationsFavoritesRequest,
  getFavoritesRequest,
  getFavoritesResponse,
  getFavoritesError,
};
