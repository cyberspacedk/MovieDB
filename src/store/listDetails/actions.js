const getCreatedListError = () => ({
  type: 'GET_LIST_DETAILS_ERROR',
});

const getListDetailsRequest = listId => ({
  type: 'GET_LIST_DETAILS_REQUEST',
  payload: listId,
});

const getListDetailsResponse = listDetails => ({
  type: 'GET_LIST_DETAILS_RESPONSE',
  payload: listDetails,
});
export { getCreatedListError, getListDetailsRequest, getListDetailsResponse };
