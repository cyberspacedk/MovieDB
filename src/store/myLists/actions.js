const createListRequest = (name, description, formikBag) => ({
  type: 'CREATE_LIST_REQUEST',
  payload: { name, description },
  formikBag,
});

const deleteListRequest = id => ({
  type: 'DELETE_LIST_REQUEST',
  payload: id,
});
const getCreatedListRequest = (page = 1) => ({
  type: 'GET_CREATED_LIST_REQUEST',
  payload: page,
});

const getCreatedListResponse = list => ({
  type: 'GET_CREATED_LIST_RESPONSE',
  payload: list,
});

const addMovieToListRequest = (listId, movieId) => ({
  type: 'ADD_MOVIE_TO_LIST_REQUEST',
  payload: { listId, movieId },
});

const getCreatedListError = () => ({
  type: 'GET_CREATED_LIST_ERROR',
});

export {
  createListRequest,
  getCreatedListRequest,
  getCreatedListResponse,
  getCreatedListError,
  deleteListRequest,
  addMovieToListRequest,
};
