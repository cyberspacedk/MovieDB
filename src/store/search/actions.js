const searchRequest = (query, page = 1) => ({
  type: 'SEARCH_REQUEST',
  payload: query,
  page,
});

const searchError = () => ({ type: 'SEARCH_ERROR' });

const searchSuccess = data => ({
  type: 'SEARCH_RESPONSE',
  payload: data,
});

export { searchRequest, searchError, searchSuccess };
