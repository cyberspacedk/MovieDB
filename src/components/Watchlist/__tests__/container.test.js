import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';

import WishlistContainer from '../container';
import { getWatchListRequest } from '../../../store/watchList/actions';

describe('Wishlist container', () => {
  const store = configureStore()({
    watchlist: {
      ids: [],
      loading: false,
      error: null,
    },
    database: {
      movies: {},
    },
  });
  store.dispatch = jest.fn();

  const wrapper = shallow(<WishlistContainer store={store} />);
  const container = wrapper.dive().dive();
  const instance = container.instance();

  it('should match snapshot', () => {
    expect(container).toMatchSnapshot();
  });

  describe('componentDidMount()', () => {
    it('should call getWatchListRequest action', () => {
      instance.componentDidMount();

      expect(store.dispatch).toHaveBeenCalledWith(getWatchListRequest());
    });
  });
});
