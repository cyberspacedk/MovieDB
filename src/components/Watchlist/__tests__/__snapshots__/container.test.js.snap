// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WatchListContainer container should match snapshot 1`] = `
<WatchList
  currentPage={0}
  empty={true}
  error={null}
  getWatchListRequest={[Function]}
  goToNextPage={[Function]}
  history={
    Object {
      "push": [Function],
    }
  }
  loading={false}
  operationsWatchListRequest={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "payload": 1,
              "type": "GET_WATCHLIST_REQUEST",
            },
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      },
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  totalResults={0}
  watchList={Array []}
/>
`;
